### DISCUSSION TITLE
Active-Passive Failover to different coordinator

### DISCUSSION THREAD

**Discussion by tabnul** [Koenkk/zigbee2mqtt#13217] [General]
I've been using kubernetes for my homesetup over the last years and it works fine for getting the software components highly available (longhorn for storage redundancy)

One thing keeps failing: i have two coordinators (CC1352P-2) and want zigbee2mqtt to be able to failover between one another (single pod)

Sometimes this  works, but sometimes it does not.
All kinds of errors , timeouts on devices etc.
It seems that the coordinator somehow keeps online / joined in the network physically even though another has been mounted to zigbee2mqtt.
I had this issue also when using VMs and CC2531 coordinators.

I think this is something hardware or firmware related. It seems the device does not shutdown / unregisters from the network physically.

Does someone recognize this issue?

I am quite sure this can be fixed by a powerdown of the usb device when not connected to z2m, since disconnecting the passive device immediately fixes the timeouts an issues

**Comment by nlahmi** [comment]
I'm not sure about the issue you're having, but I have another idea (which I intend to try and implement myself) - have each coordinator attached to a different physical node, and set the deployment so you that have one replica at any given time, only on nodes that have a coordinator (this can be setup with manual labels or with NFD probably). 
That way it also covers you in case a node completely fails and also avoids possible complications of setting up failover in Z2M. Theoretically.

**Reply by tabnul** [reply]
> I'm not sure about the issue you're having, but I have another idea (which I intend to try and implement myself) - have each coordinator attached to a different physical node, and set the deployment so you that have one replica at any given time, only on nodes that have a coordinator (this can be setup with manual labels or with NFD probably). That way it also covers you in case a node completely fails and also avoids possible complications of setting up failover in Z2M. Theoretically.

This is exactly what i do. It worked ok with the old cc2531 coordinator.

But with the CC1352P-2, after moving the z2m workload back and forth a few times, the network becomes unstable. pings timeout, commands delay or dont work at all. To me it seems.that then inactive coordinator stays physically on the network, probably because it stays powered on effectively. 

This is exactly why i wrote this issue.

**Reply by nlahmi** [reply]
Oh I see.. Did you end up solving it? (un)fortunately that is not the coordinator I have so I don't have much help to offer

**Reply by tabnul** [reply]
> Oh I see.. Did you end up solving it? (un)fortunately that is not the coordinator I have so I don't have much help to offer

Nope, didnt solve it. My assumption is that the coordinator stays powered on and receiving messages when the other host takes over , this is why i posted it  here in the firmware section.

By the way i am considering moving to one of the network attached coordinators. This woukd make using k8s easier and maybe these coordinators behave different.


I dont know what coordinator you have , just give it a try. Failover a few times and see if it stays stable.

**Reply by nlahmi** [reply]
I have the Conbee II, currently having a weird bug when doing USB Passthrough in proxmox - it works on my original HA VM, but on a k8s node it just won't get mapped to /dev/ttyACP0 (or any other indication that it works) but I guess that's a bit off topic. 
Might also give a network attached coordinator a shot

**Reply by tabnul** [reply]
> I have the Conbee II, currently having a weird bug when doing USB Passthrough in proxmox - it works on my original HA VM, but on a k8s node it just won't get mapped to /dev/ttyACP0 (or any other indication that it works) but I guess that's a bit off topic. Might also give a network attached coordinator a shot

I also use proxmox, usb passthrough to a ubuntu vm, mounted in a Z2M pod. No issues, adid le from that i have to restart the VM when i detach/reattach the usb device.
